### PR TITLE
fix: astro dev

### PR DESCRIPTION
**Pull request by brenelz**
Closes part of https://github.com/denoland/deno/issues/16659

So after a lot of debugging it seems like Astro sends a UIntArray of no length which seems to break the ReadableStream causing the server to just hang.

I am not sure if this is the right place to do this check but node must do it at some place.

I used this command to test `deno run -A --node-modules-dir npm:astro dev` in a brand new astro project with no changes.

Also something broke Astro in this commit (https://github.com/denoland/deno/commit/efcb93f8b9610bff896f21ecb5add7d17de40156) which Is why I branched before that

### CODE REVIEW

**Comment by bartlomieju** [comment]
> Also something broke Astro in this commit (https://github.com/denoland/deno/commit/efcb93f8b9610bff896f21ecb5add7d17de40156) which Is why I branched before that

Can you show the problem that this PR caused? How can I repeat it locally?

File: ext/web/06_streams.js [hunk 1]
```diff
@@ -5384,6 +5384,9 @@
      * @returns {void}
      */
     enqueue(chunk = undefined) {
+      if (chunk.byteLength === 0) {
+        return;
+      }
```
Comment (bartlomieju) [inline ext/web/06_streams.js:5389]: This seems iffy, I think we might have to fix something in std/node instead, but let's see if WPT pass

Comment (brenelz) [reply ext/web/06_streams.js:5389]: yeah i was pretty sure this isn't necessarily the right place but somehow node handles it

Comment (bartlomieju) [reply ext/web/06_streams.js:5389]: Does Astro actually use Web streams instead of Node's "readable stream"?

**Comment by brenelz** [comment]
> > Also something broke Astro in this commit ([efcb93f](https://github.com/denoland/deno/commit/efcb93f8b9610bff896f21ecb5add7d17de40156)) which Is why I branched before that
> 
> Can you show the problem that this PR caused? How can I repeat it locally?

deno run -A npm:create-astro
../deno/target/debug/deno run -A --node-modules-dir npm:astro dev

Gets a "Node.js v16.17.0 is not supported by Astro!" error. I believe because it fails this supports-esm check

https://github.com/withastro/astro/blob/main/packages/astro/astro.js#L24

**Comment by bartlomieju** [comment]
Interesting, I will debug it

File: ext/web/06_streams.js (hunk 1 above)
Comment (brenelz) [reply ext/web/06_streams.js:5389]: https://github.com/withastro/astro/blob/main/packages/astro/src/runtime/server/render/page.ts#L136

Comment (brenelz) [reply ext/web/06_streams.js:5389]: I'm not too sure i kinda fumbled around with it and found this worked

Comment (bartlomieju) [reply ext/web/06_streams.js:5389]: To be honest it feels like a bug in `ReadableStream` implementation in Node... With this change WPT seem to hang right now. Will debug it further

Comment (brenelz) [reply ext/web/06_streams.js:5389]: or maybe Astro shouldn't be sending that in the first place I don't know

**Comment by bartlomieju** [comment]
> Interesting, I will debug it

Opened https://github.com/denoland/deno/pull/17111, nice find @brenelz!

File: ext/web/06_streams.js (hunk 1 above)
Comment (bartlomieju) [reply ext/web/06_streams.js:5389]: That's a good point. @brenelz do you happen to know in which situations that happen in Astro?

Comment (brenelz) [reply ext/web/06_streams.js:5389]: it seems to do it everytime even on the most basic hello world template

Comment (kt3k) [reply ext/web/06_streams.js:5389]: this seems to throw when called with `chunk === undefined`

Comment (brenelz) [reply ext/web/06_streams.js:5389]: So would doing this instead be a valid fix? `if (chunk && chunk.byteLength === 0) {`?

Comment (iuioiua) [reply ext/web/06_streams.js:5389]: Optional chaining would be a little cleaner. E.g. `chunk?.byteLength === 0`.

Comment (brenelz) [reply ext/web/06_streams.js]: seems like the wpt still hang :(

Comment (brenelz) [reply ext/web/06_streams.js]: Figured out it is this line causing the issue. It passes an empty Uint8Array. Is this valid?

https://github.com/web-platform-tests/wpt/blob/9f9c1d83e3296584cb301e6b560ddaa70abe1627/fetch/api/request/request-init-stream.any.js#L29

Comment (brenelz) [reply ext/web/06_streams.js]: So I think we can close these pr as this is the real fix
https://github.com/denoland/deno_std/pull/3062

**Pull request closed without merging**